### PR TITLE
Fix #998 - Change exit button color for message bars + remove space at the top

### LINKED ISSUE #998: The “X” button of successful/error message bars is greyed out
#### Build:
- Stage;

#### Affected Platforms:
- Windows 10, Android 9;

#### Browsers tested on:
- Firefox, latest version;

#### Prerequisites:
- Have a Private Relay account;

#### Steps to reproduce:
1. Go to https://stage.fxprivaterelay.nonprod.cloudops.mozgcp.net/ ;
2. Sign in to the Private Relay;
3. Observe the successful message bar;

#### Expected result:
- The message bar appears with no display or layout issues;

#### Actual result:
- The message bar’s “X” button presents display issues being greyed out;

#### Notes:
- Might need some navigating back and forth, because if I remember correctly it the issue appeared sometime during today, it was not present from the beginning;
- This seems to apply to all message bars;
image
image
image
image

### PR DESCRIPTION
**Pull request by codemist**
Preview:
image

Note: Retired some utility classes.

### CODE REVIEW

File: privaterelay/templates/profile.html [hunk 1]
```diff
@@ -54,11 +54,11 @@
   {% endif %}
   <div class="main-list flx flx-col jst-cntr">
   {% if messages %}
-    <div class="messages profile-messages hidden">
+    <div class="messages hidden">
         {% for message in messages %}
         <div{% if message.tags %} class="{{ message.tags }} js-notification" {% else %} class="" {% endif %}>
           <span>{{ message }}</span>
-          <button class="flx center-value dismiss js-dismiss">
+          <button class="dismiss">
```
Comment (maxxcrawford) [inline privaterelay/templates/profile.html:61]: By removing the `js-dimiss` class, I would expect something to break? Note that it's referenced [in the app.js file](https://github.com/mozilla/fx-private-relay/blob/exit-btn-color/static/js/app.js#L269-L271). You may need to keep/add back the `js-dimiss` class.

File: privaterelay/templates/profile.html [hunk 2]
```diff
@@ -54,11 +54,11 @@
   {% endif %}
   <div class="main-list flx flx-col jst-cntr">
   {% if messages %}
-    <div class="messages profile-messages hidden">
+    <div class="messages hidden">
```
Comment (maxxcrawford) [inline privaterelay/templates/profile.html:57]: Could you revise the `.hidden` class to `.is-hidden` while you're making these changes?

File: static/scss/partials/main.scss [hunk 3]
```diff
@@ -1970,6 +1962,21 @@ input:focus::placeholder {
     background-color: $color-error;
   }
 
+  .dismiss {
+    flex-direction: column;
+    justify-content: center;
+    align-items: center;
+    display: flex;
+    margin-left: 3em;
```
Comment (maxxcrawford) [inline static/scss/partials/main.scss:1970]: Could you use a token here?

**Review by maxxcrawford — CHANGES_REQUESTED** [review]
When I log out, my message is green. Can you confirm the behavior? 

_(From running this branch locally)_ 
image

File: privaterelay/templates/profile.html (hunk 1 above)
Comment (codemist) [reply privaterelay/templates/profile.html:61]: Haha yes its supposed to be green, I think I was testing the `error` state by changing the class in Inspector. The thing is... nothing broke (at least not to my knowledge) when I removed `js-dismiss`, so should I just keep it regardless?

**Comment by codemist** [comment]
@maxxcrawford: I'll be using this PR to resolve this issue as well https://github.com/mozilla/fx-private-relay/issues/999

File: privaterelay/templates/includes/messages.html [hunk 4]
```diff
@@ -4,7 +4,7 @@
 			<div{% if message.tags %} class="message-wrapper {{ message.tags }}" {% else %} class="" {% endif %}>
 				<div class="js-notification">
 					<span>{{ message }}</span>
-					<button class="flx center-value dismiss js-dismiss">
+					<button class="dismiss js-dismiss">
```
Comment (codemist) [inline privaterelay/templates/includes/messages.html:7]: You're right, this was an oversight: `js-dismiss` is added back.

File: privaterelay/templates/profile.html [hunk 5]
```diff
@@ -16,8 +16,9 @@
 {% get_social_accounts request.user as accounts %}
 
 <main id="profile-main" class=" bg-light dashboard-container " data-api-token="{{ user_profile.api_token }}">
-  {% include "includes/messages.html" %}
-
+  <div class="messages-profile">
+    {% include "includes/messages.html" %}
```
Comment (codemist) [inline privaterelay/templates/profile.html:20]: Placed the `messages` partial into a `div` for the white background on the profile page.

File: privaterelay/templates/profile.html [hunk 6]
```diff
@@ -53,18 +54,6 @@
...
-        {% for message in messages %}
-        <div{% if message.tags %} class="{{ message.tags }} js-notification" {% else %} class="" {% endif %}>
-          <span>{{ message }}</span>
-          <button class="dismiss">
-            <svg class="x-close-icon fill-current text-gray hover:text-purple-800" role="button" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 20 20"><path d="M14.348 14.849a1.2 1.2 0 0 1-1.697 0L10 11.819l-2.651 3.029a1.2 1.2 0 1 1-1.697-1.697l2.758-3.15-2.759-3.152a1.2 1.2 0 1 1 1.697-1.697L10 8.183l2.651-3.031a1.2 1.2 0 1 1 1.697 1.697l-2.758 3.152 2.758 3.15a1.2 1.2 0 0 1 0 1.698z"/></svg>
-          </button>
-        </div>
-      {% endfor %}
-    </div>
-  {% endif %}
```
Comment (codemist) [inline privaterelay/templates/profile.html:67]: I realized this `div` should not have been here, this entire snippet belongs to the `messages` partial that's already present at line 20. The `hidden` class was what was keeping it invisible.

File: static/js/app.js [hunk 7]
```diff
@@ -1,5 +1,5 @@
 function dismissNotification() {
-	const notification = document.querySelector(".js-notification");
+	const notification = document.querySelector(".messages");
```
Comment (codemist) [inline static/js/app.js:2]: This closes the entire wrapper, eliminating any remaining padding/margin space.

File: static/scss/partials/main.scss [hunk 8]
```diff
@@ -1939,7 +1939,7 @@ input:focus::placeholder {
 
 .messages {
   width: 100%;
-  padding: $spacing-lg $spacing-md 0 $spacing-md;
+  padding: $spacing-lg $spacing-md;
```
Comment (codemist) [inline static/scss/partials/main.scss:1942]: Adds bottom padding for more.. evenness??

File: static/scss/partials/main.scss [hunk 9]
```diff
@@ -1967,7 +1967,7 @@ input:focus::placeholder {
     justify-content: center;
     align-items: center;
     display: flex;
-    margin-left: 3em;
+    margin-left: $layout-md;
```
Comment (codemist) [inline static/scss/partials/main.scss:1970]: Opted for `$layout-md` instead of spacing, since it involves the placement of an element relative to another.

**Review by codemist — COMMENTED** [review]
Started a self-review for clarification.

**Comment by codemist** [comment]
Updated demo: https://drive.google.com/drive/folders/1VsumRUevvHZr_RSk_uu8oM4gzbmu6v42?usp=sharing

File: static/js/app.js (hunk 7 above)
Comment (maxxcrawford) [reply static/js/app.js:2]: I would recommend moving the `.js-notification` to keep the class name following the [naming convention docs](https://github.com/mozilla/fx-private-relay/blob/main/docs/coding-standards.md#class-naming-convention).

**Review by maxxcrawford — CHANGES_REQUESTED** [review]

File: static/scss/partials/main.scss [hunk 10]
```diff
@@ -1953,11 +1949,14 @@ input:focus::placeholder {
...
   }
 
-  path {
-    fill: $color-white;
-  }
-
   .message-wrapper {
+    position: relative;
+    width: 100%;
+    padding: 1rem;
```
Comment (maxxcrawford) [inline static/scss/partials/main.scss:1955]: Could you make this a token?

**Review by maxxcrawford — CHANGES_REQUESTED** [review]

File: static/scss/partials/main.scss (hunk 10 above)
Comment (codemist) [reply static/scss/partials/main.scss]: Done

**Review by maxxcrawford — APPROVED** [review]

**Pull request merged**